### DISCUSSION TITLE
[Idea] CPU and Memory Switch Usage support

### DISCUSSION THREAD

**Discussion by eduardomozart** [glpi-project/glpi-agent#647] [Ideas]
### Your idea

There was a "FusionInventory SNMP" tab that provided last CPU/Memory collected during the NetInventory task, but it didn't seem available on GLPI Agent anymore. It would be great if GLPI Agent could collect and show those info (especially CPU usage) on a plot for analysis, similar to Zabbix/Cacti.
<img width="955" alt="image" src="https://github.com/glpi-project/glpi-agent/assets/2974895/ab6e997e-435d-4e97-9110-8305c9fc6f6f">

**Comment by g-bougard** [comment]
Hi,

as far I know, GLPI-Agent is still collecting this data.

You should probably open an issue on GLPI project for such data integration problem.

**Reply by eduardomozart** [reply]
Hello @g-bougard,
It isn't anymore, at least, not for Switch data. It collects the CPU usage, but there's no report on GLPI UI and most GLPI Agent templates doesn't fetch these kind of data anymore. I created custom GLPI Agent MibSupport files to enhance GLPI Agent support for CPU and RAM consumption [here](https://github.com/eduardomozart/ScriptUtil/tree/master/Scripts/GLPI/Agent/SNMP) but some of my work I create a PR directly on GLPI Agent branch if they are not related to CPU/RAM usage (I'm currently enhancing Ubiquiti devices support to report SSID names instead of radio names on Network ports tab and I'll create a PR soon).
For RAM usage, it reports the total memory of the device, but it doesn't show it's comsuption. I created the [swcpuram](https://github.com/eduardomozart/swcpuram) plug-in as an attempt to restore it's data to GLPI 10 as it used to exists with FusionInventory.

**Reply by g-bougard** [reply]
Hi,

it's fine to use dedicated MibSupport module to add CPU, RAM & MEMORY when they are missing. You can also update this structure to add OID: https://github.com/glpi-project/glpi-agent/blob/fd68f16cc604b97b14cbaf59f7014e06aae4fa85/lib/GLPI/Agent/SNMP/Device.pm#L48 even if this is less efficient than MibSupport module. Of course, it's also better to update existing MibSupport modules.

So, on agent side, your idea is still supported but the support may be limited for specific devices. In that case, don't hesitate to open an issue or even a PR to fix this support for your identified cases.

Then, on server side, or you have to request GLPI team to backport that support, or use a plugin like yours.

**Reply by eduardomozart** [reply]
There's no ``MEMORY`` DB column (that stores Flash storage size) on ``glpi_networkequipments`` table, the GLPI Inventory JSON accepts it as this field was provided by FusionInventory Agent, but it's discarded as there's no column on table so it isn't provided anymore on the Network device page, which seems like a regression for me.
I didn't update those OIDs to reflect some devices because some calculations that I do on [here](https://github.com/eduardomozart/ScriptUtil/tree/master/Scripts/GLPI/Agent/SNMP) are estimates because not all devices provides CPU % or Memory % usage, so I had to use free CPU/RAM (Mio) and compare them with total device RAM, which also works as estimates (e.g. Comware switches do not provide Total device RAM through SNMP, so I had to use free RAM to estimate total RAM based on allocated free RAM, which seems to work as expected but maybe a little bit hacky for production).

**Reply by g-bougard** [reply]
As I still said, anything related to glpi (server-side) should be reported on [glpi project](https://github.com/glpi-project/glpi).

If you think some of our MibSupport modules may be updated to enhance CPU/RAM/MEMORY support (even if not supported by GLPI 10), just create a PR and I'll check if it's relevant.
